File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/DurationType.java
package jp.superwooo.chaipon.lighttimeswitcher;

import android.content.Context;

public enum DurationType {
    Short{
        @Override
        public DurationService create(Context context) {
            return new DurationService(context, this);
        }
    },
    Long{
        @Override
        public DurationService create(Context context) {
            return new DurationService(context, this);
        }
    };
    public abstract DurationService create(Context context);
}

File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/DurationType.kt
package jp.superwooo.chaipon.lighttimeswitcher

import android.content.Context

enum class DurationType {
    Short {
        override fun create(context: Context): DurationService {
            return DurationService(context, this)
        }
    },
    Long {
        override fun create(context: Context): DurationService {
            return DurationService(context, this)
        }
    };

    abstract fun create(context: Context): DurationService
}
